service/service.go

```diff
@@ -127,21 +127,67 @@ func (s *Service) CreateUser(ctx context.Context, req *pbsvc.UserRequest) (*pbsv
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	// from here on: do not return an error because we can always regenerate tokens and resend verification emails
+	// from here on: do not return a service error
+	// because we can always regenerate tokens and resend verification emails
+	// however, upon error, return success to terminate later codes
+	user.Password = ""
+	user.IsVerified = false
+	user.PermissionLevel = auth.PermissionStringMap[auth.NoPermission]
+
+	successResponseWithoutJET := &pbsvc.UserResponse{
+		Status:  &pbsvc.UserResponse_Code{Code: uint32(codes.OK)},
+		Message: codes.OK.String(),
+		User:    user,
+	}
 
-	// create unique email token
-	emailToken, err := generateSecretKey(emailTokenByteSize)
+	// build header and body for JET
+	header := &auth.Header{
+		Alg:      auth.AlgorithmMap[auth.NoPermission],
+		TokenTyp: auth.Jet,
+	}
+	createdTimestamp := time.Now().UTC()
+	expirationTimestamp, err := generateExpirationTimestamp(createdTimestamp, daysInTwoWeeks)
 	if err != nil {
-		logger.Error(consts.CreateUserTag, consts.MsgErrGeneratingEmailToken, err.Error())
+		logger.Error(consts.CreateUserTag, consts.MsgErrGenExpirationTimestamp, err.Error())
+		return successResponseWithoutJET, nil
+	}
+
+	body := &auth.Body{
+		UUID:                user.GetUuid(),
+		Permission:          auth.UserRegistration,
+		ExpirationTimestamp: expirationTimestamp.Unix(),
+	}
+
+	// generate a new secret to expire same time as token itself (secrets expire 1 week earlier)
```

Comment:
email token expires in 2 weeks, and secret expires 1 week? I think it is better to set them 2 weeks both,